test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -6,125 +6,86 @@ import (
 	"testing"
 	"time"
 
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
-
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/rand"
-	"k8s.io/client-go/rest"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	operatorv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
 )
 
-var (
-	cfg *rest.Config
-	c   client.Client
-	ctx context.Context
-)
-
 func TestOperatorFramework(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "Operator Framework E2E Suite")
-}
+	t.Parallel()
+	cfg := ctrl.GetConfigOrDie()
 
-var _ = BeforeSuite(func() {
-	cfg = ctrl.GetConfigOrDie()
 	scheme := runtime.NewScheme()
-	ctx = context.Background()
 
-	err := catalogd.AddToScheme(scheme)
-	Expect(err).ToNot(HaveOccurred())
+	utilruntime.Must(catalogd.AddToScheme(scheme))
+	utilruntime.Must(operatorv1alpha1.AddToScheme(scheme))
 
-	err = operatorv1alpha1.AddToScheme(scheme)
-	Expect(err).ToNot(HaveOccurred())
+	c, err := client.New(cfg, client.Options{Scheme: scheme})
+	utilruntime.Must(err)
 
-	c, err = client.New(cfg, client.Options{Scheme: scheme})
-	Expect(err).ToNot(HaveOccurred())
-})
-
-var _ = Describe("Operator Framework E2E", func() {
-	var catalog *catalogd.Catalog
-	BeforeEach(func() {
-		catalog = &catalogd.Catalog{
+	var operators = []*operatorv1alpha1.Operator{
+		{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: "catalog" + rand.String(10),
+				Name: "plainv0",
```

Comment:
Why not prefer to use GenerateName or random names for parallel runs?

Reply:
Looks like these names are used in a testing log message to signal we are testing for specific bundle types

Reply:
Although, I do wonder if it would be more appropriate to create a different testing struct (similar to what we do for table tests) to have the test name that is running be more descriptive. Maybe something along the lines of:
```go
{
  name: "Operator references a plain+v0 bundle"
  operator: &operatorv1alpha1.Operator{...}
}
```

Reply:
These tests are already run in parallel.
Really no need to use random names when the names are already different, because as @everettraven points out, the name is used as part of the log output.

_IF_ this test were more complicated, I'd go with what @everettraven is suggesting, but there's only two test cases. If this were to increase to 5 or more, then, yes, a more comprehensive table test structure would be appropriate.